README.md

```diff
@@ -136,6 +136,12 @@ Select the subscription that will be used for the deployment:
 azd env set AZURE_SUBSCRIPTION_ID $AZURE_SUBSCRIPTION_ID
 ```
 
+(Optionally) Set your principal name:
```

Comment:
When it's optional, we should describe why a person would want to set it - what does it get them?  When should they not set it (and what does that do?)